R/utils.R

```diff
@@ -337,6 +264,80 @@ wb_dims <- function(...) {
 
   dims
 }
+#' @rdname dims_helper
+#' @export
+dims_to_rowcol <- function(x, as_integer = FALSE) {
+
+  dims <- x
+  if (length(x) == 1 && grepl(";", x))
+    dims <- unlist(strsplit(x, ";"))
+
+  cols_out <- NULL
+  rows_out <- NULL
+  for (dim in dims) {
+    dimensions <- unlist(strsplit(dim, ":"))
+    cols <- gsub("[[:digit:]]", "", dimensions)
+    rows <- gsub("[[:upper:]]", "", dimensions)
+
+    # if "A:B"
+    if (any(rows == "")) rows[rows == ""] <- "1"
+
+    # convert cols to integer
+    cols_int <- col2int(cols)
+    rows_int <- as.integer(rows)
+
+    if (length(dimensions) == 2) {
+      # needs integer to create sequence
+      cols <- int2col(seq.int(min(cols_int), max(cols_int)))
+      rows_int <- seq.int(min(rows_int), max(rows_int))
+    }
+
+    if (as_integer) {
+      cols <- cols_int
+      rows <- rows_int
+    } else {
+      rows <- as.character(rows_int)
+    }
+
+    cols_out <- unique(c(cols_out, cols))
+    rows_out <- unique(c(rows_out, rows))
+  }
+
+  list(cols_out, rows_out)
+}
+#' @rdname dims_helper
+#' @export
+rowcol_to_dims <- function(row, col) {
```

Comment:
I don't like this reordering. Generally code should be read from top to bottom, not in any arbitrary direction.

Reply:
I know that R is not strict in this regard, but for quite some time something like this would not even compile in c++. `rowcol_to_dims()` is used in `wb_dims()` not the other way around.

Reply:
Ok makes sense, I made this change, to change the ordering in the .Rd file (and in the reference index), but it's not necessary. Reverted